yarn/history/src/main/scala/org/apache/spark/deploy/history/yarn/YarnTimelineUtils.scala

```diff
@@ -0,0 +1,720 @@
+    URI.create(s"$protocol${address}$TIMELINE_REST_PATH")
+  }
+
+  /**
+   * Create a URI to the history service. This uses the entity type of
+   * <code>YarnHistoryService.ENTITY_TYPE</code> to
+   * @param conf
+   * @return
+   */
+  def timelineWebappUri(conf: Configuration): URI = {
+    timelineWebappUri(conf, YarnHistoryService.SPARK_EVENT_ENTITY_TYPE)
+  }
+
+  /**
+   * Get the URI of a path under the timeline web UI
+   * @param conf configuration
+   * @param subpath path under the root web UI
+   * @return a URI
+   */
+  def timelineWebappUri(conf: Configuration, subpath: String): URI = {
+    val base = getTimelineEndpoint(conf)
+    new URL(base.toURL, subpath).toURI
+  }
+
+  /**
+   * Check the service configuration to see if the timeline service is enabled
+   * @return true if `YarnConfiguration.TIMELINE_SERVICE_ENABLED`
+   *         is set.
+   */
+  def timelineServiceEnabled(conf: Configuration): Boolean = {
+    conf.getBoolean(YarnConfiguration.TIMELINE_SERVICE_ENABLED,
+                    YarnConfiguration.DEFAULT_TIMELINE_SERVICE_ENABLED)
+  }
+
+  /**
+   * Get the URI to an application under the timeline
+   * (this requires the applicationID to have been used to
+   * publish entities there)
+   * @param timelineUri timeline URI
+   * @param appId App ID (really, the entityId used to publish)
+   * @return the path
+   */
+  def applicationURI(timelineUri: URI, appId: String): URI = {
+    require(appId != null && !appId.isEmpty, "No application ID")
+    require(!appId.contains("/"), s"Illegal character '/' in $appId")
+    timelineUri.resolve(s"${timelineUri.getPath()}/$appId")
+  }
+
+  /**
+   * Map an error code to a string. For known codes, it returns
+   * a description; for others it just returns the error code.
+   * 
+   * @param code error code
+   * @return a string description for error messages
+   */
+  def timelineErrorCodeToString(code: Int): String ={
+    code match {
+      case 1 => "No start time"
+      case 2 => "IO Exception"
+      case 3 => "System Filter Conflict"
+      case 4 => "Access Denied"
+      case 5 => "No Domain"
+      case 6 => "Forbidden Relation"
+      case other: Int => s"Error code $other"
+    }
+  }
+
+  /**
+   * Convert a timeline error response to a meaningful string
+   * @param error error
+   * @return text for diagnostics
+   */
+  def describeError(error: TimelinePutError): String = {
+    s"Entity ID=${error.getEntityId()}; Entity type=${error.getEntityType}" +
+    s" Error code ${error.getErrorCode}" +
+    s": ${timelineErrorCodeToString(error.getErrorCode)}"
+  }
+
+  /**
+   * Describe a put response by enumerating and describing all errors
+   * (if present. A null errors element is handles robustly)
+   * @param response response to describe
+   * @return text for diagnostics
+   */
+  def describePutResponse(response: TimelinePutResponse) : String = {
+    val responseErrs = response.getErrors
+    if (responseErrs!=null) {
```

Comment:
`responseErrs != null`

I recommend running the `scalastyle` target in sbt on your machine (`mvn install` should also run it). Jenkins does not test build / test code since it's in a separate profile, so it's not complaining about all these style violations.

(Another way would be to add a dummy commit that makes the profile enabled by default and forces the hadoop version to 2.6, just to force jenkins to run your code.)